include/deal.II/fe/fe_enriched.templates.h

```diff
@@ -0,0 +1,206 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 - 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_fe_enriched_templates_h
+#define dealii_fe_enriched_templates_h
+
+#include <deal.II/fe/fe_enriched.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace ColorEnriched
+{
+  namespace internal
```

Comment:
Is there any reason why you put only declarations in the `fe_enriched.templates.h` file? We use `.templates.h` files for definitions, i.e., implementations of templated functions that a user might want to add its own instantiations for. I don't see that here. Is the code simply restricted to the implementation in `fe_enriched.cc`? In that case, simply move these internal declarations there. Or, if there needs to be a global visibility, I would put them into `fe_enriched.h`

Reply:
Thanks for looking through. I needed the declarations in header files because some of the tests verify the functions in "fe_enriched.templates.h" individually. I didn't want them in the header file "fe_enriched.h" since they are implementation details.

Reply:
@kronbichler I made the changes you suggested. Please review them. 

I haven't yet moved the functions since i didn't get a reply regarding this. But let me know if you prefer them moved. I will make the change.